src/sample_from_set.jl

```diff
@@ -0,0 +1,100 @@
+function load_distributions_sample_from_set()
+return quote
+
+using .Distributions: Uniform, Distribution
+
+"""
+    canonical_length(P::LazySet)
+
+Get the canonical length of the box-approximation of the convex set `P`.
+
+### Inputs
+
+- `P`   - lazyset
+
+### Outputs
+
+Matrix with two columns and `n=dims(P)` rows. Each column stands for
+one dimension of `P` whereas the first column is the minimum and the second
+column the maximum value of the corresponding dimension.
+"""
+function canonical_length(P::LazySet{N}) where {N<:Real}
+    dims = dim(P)
+    x = Array{N}(undef, dims, 2)
+    for j=1:dims
+        ej = Arrays.SingleEntryVector(j, dims, 1.0)
+        x[j,:] = [-ρ(-ej, P), ρ(ej, P)]
+    end
+    return x
+end
+
+"""
+    rand(P::LazySet{N}, n_samples::Int;
+           rng::AbstractRNG=GLOBAL_RNG,
+           seed::Union{Int, Nothing}=nothing) where {N}
+
+Rejection sampling of an arbitrary LazySet `P` for which the support value
+function is defined.
+
+Draw a sample `p` from a uniform distribution of a box-overapproximation of the
+original set `P` in all `n` dimensions. The function rejects a drawn sample `p`
+and redraws as long as the sample is not contained in the original set `P`,
+i.e., `p ∉ P`.
+
+### Input
+
+- `P`           -- lazyset
+- `n_samples`    -- number of random samples
+- `rng`         -- (optional, default: `GLOBAL_RNG`) random number generator
+- `seed`        -- (optional, default: `nothing`) seed for reseeding
+
+### Output
+
+A vector of `n_samples` vectors.
+"""
+function Base.rand(P::LazySet{N}, n_samples::Int;
```

Comment:
@mforets does not like `rand`.

Reply:
I like the use of `rand` in julia in general because then you don't have to think about what function to use to get an instance or a sample of a specific type. But I am open to suggestions. Would `sample` better?

Reply:
Indeed, we had discussion about this and the functions are so different that keeping the names separate is better IMO, provided that:

- `rand` receives a set *type* and returns a LazySet instance -- (i.e. an infinite number of points, and you draw from the distribution on the type given some constraints...)
- `sample` receives a LazySet *instance* and returns a sample of points in the set -- (a finite number of points and you draw from a distribution which is eg. uniform over the set)

Reply:
👍
Should I import `sample` from `Distributions`? Because I got the warning message
```WARNING: both Distributions and LazySets export "sample"; uses of it in module Main must be qualified```

Reply:
Yes, you should.

Reply:
I did this. But there is still the warning.
`WARNING: import of Distributions.sample into LazySets conflicts with an existing identifier; ignored.`
I think this is due to the require in https://github.com/JuliaReach/LazySets.jl/blob/f3e5fbadee35a13c26199d72f289055ad89bf15c/src/Ball2.jl#L266

Should we move the sample function outside of the `samples.jl` file?

Reply:
I found that just doing `import .Distributions` instead of `import .Distributions: sample` works.